### PR TITLE
Explicit head_dim configuration

### PR DESCRIPTION
**Pull request by talumbau**
Explicitly set head_dim in AttentionConfig

Based on @mbrenon's https://github.com/google-ai-edge/ai-edge-torch/pull/87 this PR adds head_dim configuration for

 - CrossAttention modules
 - T5 AttentionConfigs
 - StableDiffusion

BUG=https://b.corp.google.com/issues/352144652

### CODE REVIEW

File: ai_edge_torch/generative/utilities/stable_diffusion_loader.py [hunk 1]
```diff
@@ -743,6 +743,10 @@ def load(
 
     attention_config = layers_config.AttentionConfig(
```
Comment (yichunk) [inline ai_edge_torch/generative/utilities/stable_diffusion_loader.py:744]: attnetion_config should have the same update as the one in diffusion.py.
Though in the model loader, the `head_dim` is never used, so the variable here doesn't matter.
Either add a comment for reminding this is not the actual config used in the model, or update to per layer config here

**Review by yichunk — COMMENTED** [review]
LGTM for stable diffusion part.

File: ai_edge_torch/generative/utilities/stable_diffusion_loader.py (hunk 1 above)
Comment (haozha111) [reply ai_edge_torch/generative/utilities/stable_diffusion_loader.py:744]: +1

**Review by haozha111 — DISMISSED** [review]

File: ai_edge_torch/generative/utilities/stable_diffusion_loader.py (hunk 1 above)
Comment (talumbau) [reply ai_edge_torch/generative/utilities/stable_diffusion_loader.py]: Thanks @yichunk. I have just now fixed this with an additional commit. @haozha111 if you can LGTM again I willl merge in these changes.

**Pull request merged**